### ISSUE TITLE
virtio-blk not work in freebsd guest with qemu>=7.0.0

### ISSUE THREAD

**Issue by faicker** [spdk/spdk#3327]
# Sighting report

virtio-blk device  with vhost-user/spdk as BE can't be used in freebsd 14.0 guest.

## Expected Behavior

can be used.

## Current Behavior
In qemu 8.2.0, when a virtio-blk device with vhost-user/spdk as bE is attached to freebsd 14.0 guest,
the guest show errors,
`host requires unsupported maximum segment size feature.
attach returned 45`

the error is from,
[https://github.com/freebsd/freebsd-src/blob/release/14.0.0/sys/dev/virtio/block/virtio_blk.c#L348](https://github.com/freebsd/freebsd-src/blob/release/14.0.0/sys/dev/virtio/block/virtio_blk.c#L348)

qemu add the feature VIRTIO_BLK_F_SIZE_MAX from the commit [https://github.com/qemu/qemu/commit/0a963af3e3c12074ed93babe0b908a1bce79f84f](https://github.com/qemu/qemu/commit/0a963af3e3c12074ed93babe0b908a1bce79f84f)

from spdk,
[https://github.com/spdk/spdk/blob/master/lib/vhost/vhost_blk.c#L1524](https://github.com/spdk/spdk/blob/master/lib/vhost/vhost_blk.c#L1524)
 
the max is,
#define SPDK_BDEV_LARGE_BUF_MAX_SIZE (64 * 1024)

But the freebsd 14.0 maxphys default is 1M.
[https://github.com/freebsd/freebsd-src/commit/cd8537910406e68d4719136a5b0cf6d23bb1b23b](https://github.com/freebsd/freebsd-src/commit/cd8537910406e68d4719136a5b0cf6d23bb1b23b)

**Comment by changpe1** [comment]
@faicker I don't think this is a SPDK bug, but I'm OK to workaround it in SPDK, e.g: we may introduce a flag not to enable `VIRTIO_BLK_F_SIZE_MAX` feature bit for FreeBSD VM.

**Comment by faicker** [comment]
Maybe qemu should export a cmdline flag to disable the VIRTIO_BLK_F_SIZE_MAX.
I'll post to the qemu issues.

**Comment by xjjx** [comment]
@faicker are there any news? Did you have a chance to post issue to QEMU?

**Comment by faicker** [comment]
Sorry for late post.
The issue on qemu is here,
[https://gitlab.com/qemu-project/qemu/-/issues/2289](https://gitlab.com/qemu-project/qemu/-/issues/2289)

**Comment by Nikiforov-Aleksey** [comment]
Solution that worked for me: Set kern.maxphys="65536" in the guest OS.

At boot loader: set kern.maxphys="65536" then boot.

To make it permanent: echo 'kern.maxphys="65536"' >> /boot/loader.conf

This resolved the ENOTSUP (45) error and allowed the system to boot.
